### ISSUE TITLE
Feature request: `-o, --output` option to control output file name

### ISSUE THREAD

**Issue by codedump** [prjemian/spec2nexus#269]
Hello,

I'm using a directory layout where my experimental data and the analysis I make are *not* in the same directory. A `-o` option to directly control the output file name, instead of auto-generating it from the input filename, would come in handy.

I can provide code, please let me know it it's possible to intergrate it into `spec2nexus`.

Cheers,
F.

PS: Here's how it looks like based on today's HEAD: https://github.com/codedump/spec2nexus/tree/feature/output-filename

**Comment by prjemian** [comment]
Thanks for the feature request!  Should be possible.  If the desired directory path does not exist, *spec2nexus* should fail rather than attempt to create the path.  Same if the path is not writable.  Agree?

**Comment by codedump** [comment]
I'm not sure if you saw the edit of my message (I think emails don't reflect this), but I already implemented this and can put in a PR if you want (see https://github.com/codedump/spec2nexus/tree/feature/output-filename).

As to the directory part: fail is ok. I actually went with the default i.e. whatever the HDF5 writer does. I'm assuming it will fail, but haven't tried it. And failing is actually on-par with how most software out there handles these things... ¯\\_(ツ)_/¯
